core/tasty.cabal

```diff
@@ -70,8 +70,13 @@ library
     transformers         >= 0.5,
     tagged               >= 0.5,
     optparse-applicative >= 0.14,
-    unbounded-delays     >= 0.1,
     ansi-terminal        >= 0.9
+
+  -- No reason to depend on unbounded-delays on 64-bit architecture
+  if(!arch(x86_64) && !arch(aarch64))
```

Comment:
This looks brittle.
What are the possible values of `arch`?  I couldn't even find the list in the cabal docs...
At least, checking it this way (with a whitelist) is conservative.   But new architectures might easily fall under the radar.

Reply:
Full list is here: https://hackage.haskell.org/package/Cabal-3.8.1.0/docs/src/Distribution.Simple.PreProcess.html#local-6989586621679281986

Potentially `powerpc64`, `ia64` and `s390x` are also guaranteed to be 64-bit, but I decided to whitelist only architectures I have access to.

Reply:
I suggest to open an issue to potentially extend the list of recognized architectures, and whoever has access to one of them and can do testing can attack it.